### PR TITLE
Frontend updates for v0.3.0

### LINKED ISSUE #305: Support Enums for integration fields
## Tell us about the problem you're trying to solve
* Some integration configurations have fields for which the valid values can be enumerated. 

## Describe the solution you’d like
* This is supported in jsonschema and our validator. Before we can meaningfully add these enums to these configurations, we should add UI support where the UI can detect the `enum` field for a property and displays a dropdown of valid values.

## Checklist
- [ ] *Add enum support to the UI* (frontend)
- [ ] *Add enum descriptions to existing integrations (as of now, I think that is just the exchangerateapi integration)* (backend)

### LINKED ISSUE #662: Misleading error message when credentials are invalid
## Current Behavior
* When credentials are put into a source or destination, and those credentials pass the form validation but are not actually valid to connect to the integration, we display a "received form did not pass validation" error. This is misleading, we should "could not connect with provided credentials" (or something like it)

Screen Shot 2020-10-20 at 4 24 38 PM

## Steps to Reproduce
1. create postgres source
1. put in valid credentials but put in the wrong password.

### LINKED ISSUE #701: Toggle does not render properly for boolean fields.
## Steps to Reproduce
1. Attempt to use the salesforce source
1. look at is_sandbox field

Screen Shot 2020-10-26 at 9 52 30 AM

### LINKED ISSUE #702: Enums do not render choices
## Expected Behavior
When a field for an integration spec is marked as enum, the user should choose from a dropdown or radio buttons of the allowed values in that enum

you can see this in the salesforce source config in the api_type field. it is an enum that takes in REST or BULK only but the user is just presented with a text field.
Screen Shot 2020-10-26 at 9 52 37 AM

### PR DESCRIPTION
**Pull request by jamakase**
## What
- Support Enums for integration fields
- Fix toggle view
- Update error message when credentials are invalid

### CODE REVIEW

**Review by michel-tricot — APPROVED** [review]

File: airbyte-webapp/src/locales/en.json [hunk 1]
```diff
@@ -59,7 +59,7 @@
   "form.credentials_json": "Credentials JSON",
   "form.disable_collection": "Disable Collection",
   "form.replication_method": "Replication Method",
-  "form.validationError": "The received form did not pass validation",
```
Comment (cgardens) [inline airbyte-webapp/src/locales/en.json:62]: does any validation happen on the frontend? is it possible that we could actually have a form not pass validation? if it's not possible now that's fine and the backend can do it. we will then need to return different errors on the backend whether it's a validation error or an actual error.

**Review by cgardens — APPROVED** [review]
looks good!

File: airbyte-webapp/src/locales/en.json (hunk 1 above)
Comment (Jamakase) [reply airbyte-webapp/src/locales/en.json:62]: As for validation we currently validate only for required. Probably, if validation schema is provided from backend, we can also support validation on the frontend for min|max, regex and etc.

We return different types of errors based on status returned from the backend.

Comment (cgardens) [reply airbyte-webapp/src/locales/en.json:62]: great!

**Pull request merged**